src/blocker.cr

```diff
@@ -52,10 +52,21 @@ class Castblock::Blocker
 
   private def watch_device(device : Chromecast::Device, continue : Channel(Nil)) : Nil
     @chromecast.start_watcher(device, continue) do |message|
-      Log.debug &.emit("Received message", application: message.application.display_name)
+      if application = message.application
+        Log.debug &.emit("Received message", application: application.display_name)
+        if application.display_name.downcase == "youtube" && (media = message.media) && media.player_state == "PLAYING"
+          handle_media(device, media)
+        end
+      end
 
-      if message.application.display_name.downcase == "youtube" && (media = message.media) && media.player_state == "PLAYING"
-        handle_media(device, media)
+      if @mute_ads && (payload = message.payload_data)
```

Comment:
I just noticed we should add here a check `&& payload.status.size > 0`, to be sure to not crash.

Reply:
Thanks I just got the crash for this which takes a day or two to happen haha